quic/s2n-quic-core/src/varint/tests.rs

```diff
@@ -68,3 +68,87 @@ sequence_test!(four_byte_sequence_test(
+        let second_byte: u8 = kani::any();
+        // The s2n-quic implementation always chooses the smallest encoding possible.
+        // This means if we wish to test two-byte sequences, we need to encode a number
+        // that is > 63.
+        kani::assume(second_byte > 63);
+        let byte_sequence = [first_byte, second_byte];
+        let expected_val: u64 = (((first_byte & 0x3f) as u64) << 8) | (second_byte as u64);
+        assert!(expected_val <= 16383);
+        let expected = VarInt::new(expected_val).unwrap();
+        let actual_bytes = assert_codec_round_trip_value!(VarInt, expected);
+        assert_eq!(&byte_sequence[..], &actual_bytes[..]);
+    }
+
+    #[cfg(kani)]
+    #[kani::proof]
+    fn kani_four_byte_sequence_test() {
+        // https://www.rfc-editor.org/rfc/rfc9000#section-16
+        // Four byte sequences have the first two MSBs encoded as 10; the
+        // last 30 bits are the usable values.
+        let first_byte: u8 = (kani::any::<u8>() & 0x3f) | 0x80;
+        let second_byte: u8 = kani::any();
+        // The s2n-quic implementation always chooses the smallest encoding possible.
+        // This means if we wish to test the four-byte sequences, we need to encode a number
+        // that is > 16383 or 0b0011 1111 1111 1111.
+        let third_byte: u8 = kani::any();
+        kani::assume(third_byte > 0x3f);
+        let byte_sequence = [first_byte, second_byte, third_byte, 0xff];
+        let expected_val: u64 = (((first_byte & 0x3f) as u64) << 24) 
+            | (second_byte as u64) << 16 | (third_byte as u64) << 8 | 0xff;
+        assert!(expected_val <= 1073741823);
+        let expected = VarInt::new(expected_val).unwrap();
+        let actual_bytes = assert_codec_round_trip_value!(VarInt, expected);
+        assert_eq!(&byte_sequence[..], &actual_bytes[..]);
+    }
+
+    #[cfg(kani)]
+    #[kani::proof]
+    fn kani_eight_byte_sequence_test() {
+        // https://www.rfc-editor.org/rfc/rfc9000#section-16
+        // Eight byte sequences have the first two MSBs encoded as 11; the
+        // last 62 bits are the usable values.
+        let first_byte: u8 = (kani::any::<u8>() & 0x3f) | 0xc0;
+        let second_byte: u8 = kani::any();
+        // The s2n-quic implementation always chooses the smallest encoding possible.
+        // This means if we wish to test eight-byte sequences, we need to encode a number
+        // that is > 1073741823 or 0b0011 1111 1111 1111 1111 1111 1111 1111
+        let third_byte: u8 = kani::any();
+        let fourth_byte: u8 = kani::any();
+        kani::assume(fourth_byte > 0x3f);
+        let byte_sequence = [first_byte, second_byte, third_byte, fourth_byte, 0xff, 0xff, 0xff, 0xff];
+        let expected_val: u64 = (((first_byte & 0x3f) as u64) << 56) 
+            | (second_byte as u64) << 48 | (third_byte as u64) << 40 | (fourth_byte as u64) << 32 
+            | (0xff << 24) | (0xff << 16) | (0xff << 8) | 0xff;
```

Comment:
it's a bit easier to read if you convert this value from a BE array
```suggestion
        let expected_val = u64::from_be_bytes([first_byte & 0x3f, second_byte, third_byte, fourth_byte, 0xff, 0xff, 0xff, 0xff]);
```